eng/pipelines/templates/steps/install-dotnet.yml

```diff
@@ -1,6 +1,7 @@
 steps:
   - task: UseDotNet@2
     displayName: 'Use .NET Core SDK'
+    condition: ne(variables['Agent.OS'], 'Windows_NT') # Windows supports MultiLevelLookup and doesn't need explicit framework installation
```

Comment:
We should update these condtionds to include `and(succeeded(), ....` otherwise in a failed or canceled build these steps will still try to execute.